src/AnalyticsEventReporter.ts

```diff
@@ -0,0 +1,36 @@
+import { AnalyticsEventService } from "./AnalyticsEventService"
+import { AnalyticsConfig } from "./AnalyticsConfig";
+import { EventPayload } from "./EventPayload";
+import { EventAPIResponse } from "./EventAPIResponse";
+
+/** Represents an reporter is responsible for reporting analytics events. */
+export class AnalyticsEventReporter implements AnalyticsEventService {
+
+    private config: AnalyticsConfig;
+    private payload?: EventPayload;
+
+    /**
+     * @param config - necessary analytics config: Must provide one and only
+     * one of API Key or Bearer Token.
+     * 
+     * @param payload - (optional) desired event values to report
+     */
+    constructor(config: AnalyticsConfig, payload?: EventPayload) {
+        const apiKeyIsSet = typeof config.key === 'string' && config.key.trim().length !== 0;
```

Comment:
There is no need to check if it is a string. If you are doing this for the sake of confirming it is neither undefined or null, you can just use `??`, which functions exactly the same as coalesce in mySQL.

I would also not bother with trimming the whitespace and validating if they are putting in a string of spaces. If the user does that, that's on them.

Reply:
Actually, how about config.key !== undefined? It considers cases when key is not set in config and when key is specified as undefined.

Reply:
That seems fine to me.